test/extended/networking/ipsec.go

```diff
@@ -632,6 +653,117 @@ var _ = g.Describe("[sig-network][Feature:IPsec]", g.Ordered, func() {
 	})
 })
 
+var _ = g.Describe("[sig-network] IPsec resilience", g.Ordered, func() {
+	oc := exutil.NewCLIWithPodSecurityLevel("ipsec", admissionapi.LevelPrivileged)
+	f := oc.KubeFramework()
+	var ipsecMode v1.IPsecMode
+
+	InOVNKubernetesContext(func() {
+		g.BeforeAll(func() {
+			var err error
+			ipsecMode, err = getIPsecMode(oc)
+			o.Expect(err).NotTo(o.HaveOccurred())
+		})
+
+		g.It("check pod traffic are working across nodes", func() {
+			if ipsecMode != v1.IPsecModeFull {
+				e2eskipper.Skipf("cluster is configured with IPsec %s mode, so skipping the test", ipsecMode)
+			}
+			g.By("creating test pods")
+			pods := createWebServerPods(oc, f.Namespace.Name)
+			g.By("checking crossing connectivity over the pods")
+			checkPodCrossConnectivity(pods)
+		})
+
+		g.It("check pod traffic are working across nodes after ipsec daemonset restart", func() {
+			if ipsecMode != v1.IPsecModeFull {
+				e2eskipper.Skipf("cluster is configured with IPsec %s mode, so skipping the test", ipsecMode)
+			}
+			g.By("creating test pods")
+			pods := createWebServerPods(oc, f.Namespace.Name)
+			g.By("checking crossing connectivity over the pods")
+			checkPodCrossConnectivity(pods)
+			// Restart IPsec daemonset few times and check pod traffic is not impacted.
+			for i := 1; i <= 5; i++ {
+				g.By(fmt.Sprintf("attempt#%d restarting IPsec pods", i))
+				err := restartIPsecDaemonSet(oc)
+				o.Expect(err).NotTo(o.HaveOccurred())
```

Comment:
nit; i like to see the reasons why something failed with this expects otherwise you have to go look at the line of code that it failed on.

Reply:
oh! yes martin, good catch. added `restartIPsecDaemonSet` as a test helper function, that will solve the problem that you described.